bundle/edu.gemini.p2checker/src/test/scala/edu/gemini/p2checker/rules/GmosSDtaXRuleTest.scala

```diff
@@ -56,31 +56,45 @@ class GmosSDtaXRuleTest extends AbstractRuleTest {
     rule.check(elems).getProblems.asScala.toList
   }
 
-  private def expectSuccess(): Unit = {
-//    println(runCheck.mkString(",\n"))
-    assertTrue(runCheck.isEmpty)
+  private def expectSuccess(hasId: Set[String]): Unit =
+    assertTrue(!runCheck.exists(p => hasId(p.getId)))
+
+  private def expectSuccessBinY(): Unit =
+    expectSuccess(Set(
+      GmosRule.GMOS_S_DTA_X_RULE_Y1_ID,
+      GmosRule.GMOS_S_DTA_X_RULE_Y2_4_ID)
+    )
+
+  private def expectSuccessBinMultiple(): Unit = {
+    assertTrue(!runCheck.exists { p =>
+      GmosRule.DTA_X_Y_MULTIPLE_BINNING_RULE_ID == p.getId()
+    })
   }
-
   private def expectFailure(ybin: String): Unit =
     assertTrue(runCheck.exists { p =>
       p.getType == Problem.Type.ERROR &&
       p.getDescription.contains(ybin)
     })
```

Comment:
I'll also remove the current parameter that checks for a `String` in the error message and instead check for the problem ID, just like when checking for success.